### PR TITLE
Fixes #481 Spotlight | Custom scroll targets

### LINKED ISSUE #481: Spotlight, useScrollObserver | allow for custom scroll targets
Currently the useScrollObserver and Spotlight both use the window while detecting/setting the scroll position. This doesn't work in cases where a div or other element is what scrolls on the page.

- [x] Add prop for Spotlight to allow for custom scrolling element
- [x] Add argument in useScrollObserver to observe a custom scrolling target
- [x] Include a new storybook story that showcases the new spotlight behavior

### PR DESCRIPTION
**Pull request by aVileBroker**

### CODE REVIEW

File: src/components/Spotlight/Spotlight.stories.tsx [hunk 1]
```diff
@@ -109,7 +109,137 @@ export const AnimatedSpotlight: Story = (args: Partial<SpotlightProps>) => {
...
+  const [tourStarted, setTour] = useState<boolean>(false);
+
+  const stepOptions = [null, menuRef, cardRef, buttonRef];
+  const messages = [
+    {
+      title: 'Welcome to the tutorial!',
+      subtitle: 'targetElement is null for this part of the tour, so nothing is highlighted!',
+    },
+    { title: 'This is a kebab menu.', subtitle: '' },
+    { title: 'This is the whole card', subtitle: '' },
```
Comment (dawsonbooth) [inline src/components/Spotlight/Spotlight.stories.tsx:177]: Some punctuation for consistency

```suggestion
    { title: 'This is the whole card.', subtitle: '' },
```

File: src/components/Spotlight/Spotlight.tsx [hunk 2]
```diff
@@ -120,6 +144,14 @@ const Spotlight = ({
 
   const [isAutoScrolling, setIsAutoScrolling] = useState(false);
 
+  useEffect(() => {
+    if (targetElement && !scrollTarget.current) {
+      scrollTarget.current = findNearestScrollingParent(targetElement);
+    }
+  }, [targetElement]);
+
+  console.log(scrollTarget.current);
```
Comment (dawsonbooth) [inline src/components/Spotlight/Spotlight.tsx:153]:
```suggestion
```

File: src/components/Spotlight/Spotlight.tsx [hunk 3]
```diff
@@ -129,15 +161,33 @@ const Spotlight = ({
       // (if the annotation is below the target, this should be an addition, not subtraction)
       const offset = annotationHeight ? newTargetTop - annotationHeight : newTargetTop;
 
-      window.scrollBy({
-        top: offset,
-        left: 0,
-        behavior: !animateTargetChanges || prefersReducedMotion ? 'auto' : 'smooth',
-      });
+      if (scrollTarget.current) {
+        // TODO:
```
Comment (dawsonbooth) [inline src/components/Spotlight/Spotlight.tsx:165]: Is this still a TODO or has it been TODone?

**Review by dawsonbooth — CHANGES_REQUESTED** [review]

File: src/components/Spotlight/Spotlight.tsx (hunk 3 above)
Comment (aVileBroker) [reply src/components/Spotlight/Spotlight.tsx:164]: Still a TODO, I'll make a ticket

Comment (aVileBroker) [reply src/components/Spotlight/Spotlight.tsx]: Done 👍

**Review by dawsonbooth — APPROVED** [review]

**Review by jonesdebu — APPROVED** [review]

**Pull request merged**